job/server/src/main/java/alluxio/master/job/metrics/DistributedCmdMetrics.java

```diff
@@ -229,40 +179,47 @@ private static String getFilePathForNonBatchConfig(JobConfig config) {
   public static void batchIncrementForCompleteStatusWithRetry(
```

Comment:
Is metrics.inc() a bottle neck here so we have two separate function? if not can we just use incrementForCompleteStatusWithRetry in batch situation to avoid duplicate code?

Reply:
The BatchConfig has `final Set<Map<String, String>> mJobConfigs` as its job configs, it would not be easy to cast a map to a JobConfig type directly, so I use two functions for batch and non-batch cases.

Reply:
check out `selectExecutors` in BatchedJobDefinition for casting

Reply:
I see, thanks. But if we use incrementForCompleteStatusWithRetry in batch situation, we would need to iterate on checking jobType for each jobConfig, and do counter increments like JOB_DISTRIBUTED_LOAD_SUCCESS.inc(1) , would that be time consuming as well? Currently for batch job, we just need one jobType check, and one JOB_DISTRIBUTED_LOAD_SUCCESS.inc(count) will be much faster.

Reply:
yeah that's the reason I'm asking whether metrics.inc() is a bottleneck here(casting should be quick).  I have no strong opinion on this and would leave it up to you.

Reply:
Oh I see, thanks. Yes, that will be the bottleneck when batch size is very big. I'll leave this unchanged for now, and look at this later when i have time.